site/content/blog/simplifying-package-authoring.md

```diff
@@ -0,0 +1,301 @@
+2. Identify K8s manifest which needs to be packaged. [`Dynatrace Operator`](https://github.com/Dynatrace/dynatrace-operator) releases the [`kubernetes.yaml`](https://github.com/Dynatrace/dynatrace-operator/releases) which can be packaged and be available for distribution.
+3. K8s cluster (I will be using minikube).
+4. OCI registry where the package bundle and repository bundles will be pushed (I will be using my DockerHub account).
+
+### kctrl pkg init
+
+```bash
+$ mkdir dynatrace 
+$ cd dynatrace
+$ kctrl pkg init
+```
+
+This command asks a few basic questions regarding how we want to initialize our package. Lets go through the question together:
+
+![Simplifying package authoring - pkg init basic details](/images/blog/simplifying-package-authoring-package-basic-details.png)
+
+* In this question, we need to enter the package name which will be a valid DNS subdomain name.
+
+![Simplifying package authoring - pkg init content option](/images/blog/simplifying-package-authoring-package-content-option.png)
+
+* Here, we need to enter from where to get the K8s manifest which needs to be packages. As mentioned earlier, `dynatrace` releases `kubernetes.yaml` as part of their github release artifacts. Hence, we will select `Option 2`.
+
+![Simplifying package authoring - pkg init dynatrace repository details](/images/blog/simplifying-package-authoring-dynatrace-repository-details.png)
+
+* In this few questions are being asked to enter github slug, tag/release version and the manifest file names etc.
+
+![Simplifying package authoring - pkg init vendir sync](/images/blog/simplifying-package-authoring-vendir-sync.png)
+
+* After entering all the above details, tool will download all the mentioned manifest locally and create two files - package-build.yml and pacakge-resources.yml. `Package-resources.yml` file contain Package, PackageInstall and PackageMetadata while `package-build.yml` contains PackageBuild. More information on PackageBuild can be found in this [link](). All the values entered above have been recorded in these files. These files will be used by the `dev` and `pkg release` command subsequently. 
+
+### kctrl dev
+
+To keep the blog short, I am not going to add any additional overlay/manifest. To add them, please follow this [link]().
+
+`kctrl dev` command is used to deploy the package locally. Local means that on the K8s cluster, there is no need to install the `kapp-controller`. But still the package will be deployed in a similar manner as by the `kapp-controller` and see how the K8s resources behave. This will help authors to get a quick feedback on the changes so that users can iterate over them quickly. As the `Package` and `PackageInstall` are present in the package-resources, we will provide `package-resources.yml` file to `kctrl dev`.
```

Comment:
```suggestion
`kctrl dev` command is used to deploy the package locally. Local means that on the Kubernetes cluster, there is no need to install the `kapp-controller`. But still the package will be deployed in a similar manner as by the `kapp-controller` and see how the Kubernetes resources behave. This will help authors to get a quick feedback on the changes so that users can iterate over them quickly. As the `Package` and `PackageInstall` are present in the package-resources, we will provide `package-resources.yml` file to `kctrl dev`.
```

Reply:
Adopted